### PR TITLE
add workflow for vsphere-ipi-disk-encryption-tang

### PR DESCRIPTION
**Pull request by jinyunma**
1. add workflow for vsphere-ipi-disk-encryption-tang
2. use butane to generate tang manifest file

### CODE REVIEW

**Comment by jinyunma** [comment]
@jianlinliu @gpei PTAL, thanks.

File: ci-operator/step-registry/vsphere/provision/bastionhost/tang/vsphere-provision-bastionhost-tang-ref.metadata.json [hunk 1]
```diff
@@ -0,0 +1,12 @@
+{
+	"path": "vsphere/provision/bastionhost/tang/vsphere-provision-bastionhost-tang-ref.yaml",
```
Comment (jianlinliu) [inline ci-operator/step-registry/vsphere/provision/bastionhost/tang/vsphere-provision-bastionhost-tang-ref.metadata.json:2]: Most of implementation in the step is to generate manifest files, so maybe we  can rename it to `ipi-conf-tang`, also we can share the step with other platform in the future.

Comment (jinyunma) [reply ci-operator/step-registry/vsphere/provision/bastionhost/tang/vsphere-provision-bastionhost-tang-ref.metadata.json]: updated

File: ci-operator/step-registry/ipi/conf/tang/ipi-conf-tang-ref.yaml [hunk 2]
```diff
@@ -0,0 +1,12 @@
+ref:
+  as: ipi-conf-tang
+  from: upi-installer
```
Comment (jianlinliu) [inline ci-operator/step-registry/ipi/conf/tang/ipi-conf-tang-ref.yaml:3]: Maybe we can replace the image with ` from: cli`

Comment (jinyunma) [reply ci-operator/step-registry/ipi/conf/tang/ipi-conf-tang-ref.yaml]: In step `ipi-conf-tang`, require `jq` command, but it is not installed in cli image, so still needs `upi-installer` image here.

Comment (jianlinliu) [reply ci-operator/step-registry/ipi/conf/tang/ipi-conf-tang-ref.yaml]: Maybe we can use this one:
```
  from_image:
    namespace: ocp
    name: cli-jq
    tag: latest
```

Comment (jinyunma) [reply ci-operator/step-registry/ipi/conf/tang/ipi-conf-tang-ref.yaml]: ah, let me update to use `cli-jq`

Comment (jinyunma) [reply ci-operator/step-registry/ipi/conf/tang/ipi-conf-tang-ref.yaml]: get new error, no ssh command in image "cli-jq".
` /bin/bash: line 40: ssh: command not found `

Comment (jianlinliu) [reply ci-operator/step-registry/ipi/conf/tang/ipi-conf-tang-ref.yaml]: Per the error - `/bin/bash: line 40: ssh: command not found`, seem like cli-jq does not install ssh client, let us use:
```
  from_image:
    namespace: ocp
    name: "4.12"
    tag: upi-install
```
to replace `from: upi-installer` which has dependency on job definition.

**Pull request merged**